CRM/Contribute/BAO/Contribution.php

```diff
@@ -2312,7 +2312,7 @@ public static function transitionComponents($params, $processContributionObject
                 $membership->membership_type_id,
                 (array) $membership
               );
-              $membership->status_id = CRM_Utils_Array::value('id', $status, $membership->status_id);
+              $membership->status_id = $status['id'] ?? $membership->status_id;
               $membership->save();
```

Comment:
I think this one's ok because of the way getMembershipStatusByDate works. `id` would either be missing completely or would be a number, so the swap is valid.